src/Symfony/Component/Mailer/Bridge/Amazon/Transport/SesTransportFactory.php

```diff
@@ -11,35 +11,46 @@
 
 namespace Symfony\Component\Mailer\Bridge\Amazon\Transport;
 
+use AsyncAws\Core\Configuration;
 use Symfony\Component\Mailer\Exception\UnsupportedSchemeException;
 use Symfony\Component\Mailer\Transport\AbstractTransportFactory;
 use Symfony\Component\Mailer\Transport\Dsn;
 use Symfony\Component\Mailer\Transport\TransportInterface;
 
 /**
  * @author Konstantin Myakshin <molodchick@gmail.com>
+ * @author Jérémy Derussé <jeremy@derusse.com>
  */
 final class SesTransportFactory extends AbstractTransportFactory
 {
     public function create(Dsn $dsn): TransportInterface
     {
         $scheme = $dsn->getScheme();
-        $user = $this->getUser($dsn);
-        $password = $this->getPassword($dsn);
         $region = $dsn->getOption('region');
-        $host = 'default' === $dsn->getHost() ? null : $dsn->getHost();
-        $port = $dsn->getPort();
 
-        if ('ses+api' === $scheme) {
-            return (new SesApiTransport($user, $password, $region, $this->client, $this->dispatcher, $this->logger))->setHost($host)->setPort($port);
-        }
-
-        if ('ses+https' === $scheme || 'ses' === $scheme) {
-            return (new SesHttpTransport($user, $password, $region, $this->client, $this->dispatcher, $this->logger))->setHost($host)->setPort($port);
-        }
-
-        if ('ses+smtp' === $scheme || 'ses+smtps' === $scheme) {
-            return new SesSmtpTransport($user, $password, $region, $this->dispatcher, $this->logger);
+        switch ($scheme) {
+            case 'ses+smtp':
+            case 'ses+smtps':
+                return new SesSmtpTransport($this->getUser($dsn), $this->getPassword($dsn), $region, $this->dispatcher, $this->logger);
+            case 'ses+api':
+                $class = SesApiTransport::class;
+                // do not break
+            case 'ses':
+            case 'ses+https':
+                $class = $class ?? SesHttpTransport::class;
+                if (!class_exists(Configuration::class)) {
+                    throw new \LogicException(sprintf('You cannot use "%s" as the AsyncAws component is not installed. Try running "composer require async-aws/ses".', $class));
```

Comment:
Will this ever happen?

Reply:
yes, `async-aws` is not required, people may just use the `smtp://` transport

Reply:
Oh, yeah. I see now it is in require-dev. 

Good

Reply:
to be honest, I don't know if it worth using the amazon bridge with the `smtp` transport. One could just use the `Smtp` bridge to get the same result